File: sem iv/Theory of Computaion/lab/code/c/3_prefix_suffix_all.cpp
// Program: Display all prefixes and suffixes of a given string
#include <iostream>
#include <cstring>
using namespace std;

void prefix1(char input[100]);
void suffix1(char input[100]);

int main() {
    char input[100];
    cout << "Enter the string" << endl;
    cin.getline(input, 100);

    prefix1(input);
    suffix1(input);

    cin.get();
    return 0;
}

void prefix1(char input[100]) {
    cout << "The possible prefixes are" << endl;
    for (int i = (int)strlen(input); i > 0; i--) {
        for (int j = 0; j < i; j++)
            cout << input[j];
        cout << "\n";
    }
}

void suffix1(char input[100]) {
    cout << "The possible suffixes are" << endl;
    for (int i = 0; i < (int)strlen(input); i++) {
        for (int j = i; j < (int)strlen(input); j++)
            cout << input[j];
        cout << "\n";
    }
}

File: sem iv/Theory of Computaion/lab/code/python/3_prefix_suffix_all.py
# Program: Display all prefixes and suffixes of a given string

def prefix1(w):
    print("The possible prefixes are")
    for i in range(len(w), 0, -1):
        print(w[:i])

def suffix1(w):
    print("The possible suffixes are")
    for i in range(len(w)):
        print(w[i:])

def main():
    w = input("Enter the string\n")
    prefix1(w)
    suffix1(w)

if __name__ == "__main__":
    main()
